docs/source/schemas/streams.rst

```diff
@@ -40,6 +44,23 @@ this is to write a ``KeyValueService`` implementation that just
 delegates to 2 other ``KeyValueService`` impls and sends COLD tables to
 one and other tables to another.
 
+Transactionality
+================
+
+When you call ``loadStream(transaction, id)``, only the first section of the stream data is pre-loaded as part of that transaction.
+The amount loaded is determined by the block size (1MB) and the in-memory threshold (4MiB by default); we load at least one block,
+a whole number of blocks, and (if the in-memory threshold is at least the block size) as many blocks as fit inside the in-memory threshold.
+So by default, we load 4 blocks (``4*10^6`` bytes, slightly less than the default in-memory threshold of ``4*2^20`` bytes).
+If your stream does not inside the in-memory threshold, then the remainder of the data will be buffered in separate transactions.
+
+This has some subtle transactionality implications.
+Suppose you store a mapping of keys to stream IDs, and at ``start_timestamp``, ``key`` maps to ``stream_id_1``.
+You then call ``loadStream(transaction,stream_id_1)``, and keep the resulting ``InputStream`` open.
+Before you reach the end of the stream, another thread stores a mapping from ``key`` to ``stream_id_2``.
+In this case, streaming of the original stream will continue and complete successfully, even though the data being streamed is out of date.
```

Comment:
Hmm, trying to understand the issue here. Since `stream_id` corresponds to exactly one immutable stream, I don't see why your transactional guarantees are any worse here than if you stored the entire stream in the `key` row. It seems like you could consider `stream_id` as a compressed `stream`, and consider reading the `stream` as decompressing it.